src/Processors/Formats/Impl/ArrowColumnToCHColumn.cpp

```diff
@@ -484,19 +498,36 @@ static void checkStatus(const arrow::Status & status, const String & column_name
         throw Exception{ErrorCodes::UNKNOWN_EXCEPTION, "Error with a {} column '{}': {}.", format_name, column_name, status.ToString()};
 }
 
-Block ArrowColumnToCHColumn::arrowSchemaToCHHeader(const arrow::Schema & schema, const std::string & format_name, const Block * hint_header)
+Block ArrowColumnToCHColumn::arrowSchemaToCHHeader(
+    const arrow::Schema & schema, const std::string & format_name, const Block * hint_header, bool ignore_case)
 {
     ColumnsWithTypeAndName sample_columns;
     std::unordered_set<String> nested_table_names;
     if (hint_header)
         nested_table_names = Nested::getAllTableNames(*hint_header);
+
+    const auto accept_field = [&](const auto & field_name)
+    {
+        if (!hint_header || hint_header->has(field_name, ignore_case))
+            return true;
+
+        if (!ignore_case)
+            return nested_table_names.contains(field_name);
+
+        return std::find_if(
+                   nested_table_names.begin(),
+                   nested_table_names.end(),
+                   [&](const auto & nested_table_name) { return equalsCaseInsensitive(nested_table_name, field_name); })
+            != nested_table_names.end();
```

Comment:
I would suggest add flag `bool to_lower_case` in `Nested::getAllTableNames` to get the set of names in lower case and use just `nested_table_names.contains()` with `field_name` in lower case. Here and in the other similar places. But if you think current way is better, it's ok.